# GROUP 8


**Binary Brains**


Members
Althea Jane Trinidad
Harvey Kim Solano

Sep. 19, 2025


NiTPHLeX is a fast-growing video streaming platform managed by the Hill Group of Companies and founded by Paulo Hill. It has quickly gained popularity in Southeast Asia by focusing on regional movies, indie content, and localized subtitles, making it a preferred choice for audiences seeking culturally relevant entertainment.

In [ ]:
%pip install qdesc
%pip install researchpy
import pandas as pd
import numpy as np
import qdesc as qd
import researchpy as rp

In [ ]:
# Loading a data set and storing it into a variable.
df = pd.read_excel("NiTPHLeXVideoStreaming.xlsx")
# Checking the first three rows of the data set.
df.head(5)

user_location_type  peak_hour_flag  concurrent_users  \
0              urban            True              2173   
1              rural           False               859   
2              rural           False               207   
3              urban           False               228   
4              urban            True              1295   

   buffer_events_per_session  mobile_error_rate(proportion)  \
0                          3                           0.04   
1                          2                           0.07   
2                          2                           0.09   
3                          2                           0.03   
4                          0                           0.04   

   video_load_time(seconds)  subtitle_accuracy_score(0-1)  \
0                      6.09                          0.93   
1                      6.76                          0.96   
2                      5.15                          0.95   
3                      4.21                          0.97   
4                      7.24                          0.96   

   average_viewing_time(hours) churn_flag  server_cost(in USD)  \
0                         0.29        Yes               402.79   
1                         0.88         No               280.01   
2                         1.55         No               235.06   
3                         0.47        Yes               234.12   
4                         1.64         No               409.14   

   content_engagement_score(0-100)  
0                            75.55  
1                            75.82  
2                            76.42  
3                            77.30  
4                            75.58

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3487 entries, 0 to 3486
Data columns (total 11 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   user_location_type               3487 non-null   object 
 1   peak_hour_flag                   3487 non-null   bool   
 2   concurrent_users                 3487 non-null   int64  
 3   buffer_events_per_session        3487 non-null   int64  
 4   mobile_error_rate(proportion)    3487 non-null   float64
 5   video_load_time(seconds)         3487 non-null   float64
 6   subtitle_accuracy_score(0-1)     3487 non-null   float64
 7   average_viewing_time(hours)      3487 non-null   float64
 8   churn_flag                       3487 non-null   object 
 9   server_cost(in USD)              3487 non-null   float64
 10  content_engagement_score(0-100)  3487 non-null   float64
dtypes: bool(1), float64(6), int64(2), object(2)
memory usage: 276.0+ KB


In [ ]:
qd.desc(df)

count    mean     std  median     MAD  \
concurrent_users                 3487.0  897.73  663.50  702.00  257.00   
buffer_events_per_session        3487.0    1.30    1.29    1.00    1.00   
mobile_error_rate(proportion)    3487.0    0.06    0.03    0.06    0.02   
video_load_time(seconds)         3487.0    6.02    1.93    5.97    1.32   
subtitle_accuracy_score(0-1)     3487.0    0.93    0.07    0.95    0.02   
average_viewing_time(hours)      3487.0    1.05    0.51    1.05    0.36   
server_cost(in USD)              3487.0  320.72   65.62  314.39   43.58   
content_engagement_score(0-100)  3487.0   75.60    1.71   75.88    0.79   

                                    min      max  AD_stat  5% crit_value  
concurrent_users                 200.00  2997.00   245.07           0.79  
buffer_events_per_session          0.00     9.00   170.24           0.79  
mobile_error_rate(proportion)      0.01     0.15    55.73           0.79  
video_load_time(seconds)           0.14    13.66     2.44           0.79  
subtitle_accuracy_score(0-1)       0.49     1.03   499.57           0.79  
average_viewing_time(hours)        0.10     2.98     3.64           0.79  
server_cost(in USD)              122.83   573.89     9.11           0.79  
content_engagement_score(0-100)   65.72    79.60   122.22           0.79

**Identified Problems**

In [ ]:
df['buffer_events_per_session'].describe()
df['video_load_time(seconds)'].describe()
df['mobile_error_rate(proportion)'].describe()
df['server_cost(in USD)'].describe()
df['subtitle_accuracy_score(0-1)'].describe()
df['average_viewing_time(hours)'].describe()
df['churn_flag'].value_counts(normalize=True) * 100
df['content_engagement_score(0-100)'].describe()
df['concurrent_users'].describe()

count    3487.000000
mean      897.733295
std       663.502054
min       200.000000
25%       450.500000
50%       702.000000
75%       965.000000
max      2997.000000
Name: concurrent_users, dtype: float64

The analysis of the concurrent_users variable shows that NiTPHLeX recorded 3,487 observations, with an average of about 898 users online simultaneously. However, the high standard deviation of 663 indicates large fluctuations in user activity, ranging from as low as 200 users during off-peak hours to as high as 2,997 at peak demand. The distribution reveals that 25% of the time, concurrent users are below 451, while 75% of the time, they are below 965, meaning only the top 25% of cases exceed this level. This wide gap between the mean and the maximum highlights that while the system usually handles moderate loads, extreme spikes in concurrent users are straining the CDN infrastructure, leading to issues such as buffering, long load times, and reduced performance during peak periods.